xgi/drawing/draw.py

```diff
@@ -985,3 +987,246 @@ def _update_lims(pos, ax):
     corners = (minx - padx, miny - pady), (maxx + padx, maxy + pady)
     ax.update_datalim(corners)
     ax.autoscale_view()
+
+
+def _draw_hull(node_pos, ax, edges_ec, facecolor, alpha, zorder, radius):
+    """Draw a convex hull encompassing the nodes in node_pos
+
+    Parameters
+    ----------
+    node_pos : np.array
+        nx2 dimensional array containing positions of the nodes
+    ax : matplotlib.pyplot.axes
+    edges_ec : str
+        Color of the border of the convex hull
+    facecolor : str
+        Filling color of the convex hull
+    alpha : float
+        Transparency of the convex hull
+    radius : float
+        Radius of the convex hull in the vicinity of the nodes.
+
+    Returns
+    -------
+    ax : matplotlib.pyplot.axes
+
+    """
+
+    points = node_pos.copy()
+    for i in node_pos:
+        center = i
+        thetas = np.linspace(0, 2 * np.pi, num=100, endpoint=False)
+        circle = [
+            center + radius * np.array([np.cos(theta), np.sin(theta)])
+            for theta in thetas
+        ]
+        points = np.vstack([circle, points])
```

Comment:
I believe the following two functions achieve the same result. The first is essentially a copy paste from the highlighted code.
```python3
def foo():
    node_pos = np.random.random(size=(100, 2))
    radius = 1.5
    points = node_pos.copy()

    for i in points:
        center = i
        thetas = np.linspace(0, 2*np.pi, num=100, endpoint=False)
        circle = [center + radius * np.array([np.cos(theta), np.sin(theta)]) for theta in thetas]
        points = np.vstack([circle, points])

    return points


def foo2():
    node_pos = np.random.random(size=(100, 2))
    radius = 1.5
    thetas = np.linspace(0, 2*np.pi, num=100, endpoint=False)
    points = np.vstack([
        p + radius * np.array([np.cos(thetas), np.sin(thetas)]).T
        for p in node_pos
    ])
    return np.vstack([node_pos, points])
```

If we all agree that these indeed produce the same output, then we should use the second one, as it's 40x faster:
```python3
In [37]: %timeit foo()
40 ms ± 620 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [38]: %timeit foo2()
698 µs ± 11.9 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

```

Reply:
Hi @leotrs, thank you for the input, it is now implemented in the second way.

Reply:
Thanks!